llvm-general/test/LLVM/General/Test/Linking.hs

```diff
@@ -59,9 +59,9 @@ tests = testGroup "Linking" [
         ]      
 
     Module { moduleDefinitions = defs } <- withContext $ \context -> 
-      withModuleFromAST' context ast0 $ \m0 ->
-        withModuleFromAST' context ast1 $ \m1 -> do
-          runExceptT $ linkModules False m0 m1
-          moduleAST m0
+      withModuleFromAST' context ast0 $ \dest -> do
+        src <- failInIO $ astToFFIModule context ast1
+        failInIO $ linkModules dest src
```

Comment:
This change is necessary for 3.9 since LLVM now destroys the src module. Using `withModuleFromAST'` tries to destroy the module which causes segfaults.

I guess we could make a newtype for modules that are consumed but until we get linear types or something like that in Haskell I don’t see how we can actually protect users from reusing the module after at has been destroyed.